### PR TITLE
Removed unneeded outputs and the corresponding tests. Added postCreat…

### PR DESCRIPTION
**Pull request by svdgraaf**
As mentioned in issue https://github.com/serverless/serverless/issues/1469 We don't need the iamRoleArn property. We can have CloudFormation generate it dynamically for us, that's what this PR will do :smiley:

This is the squashed commit for https://github.com/serverless/serverless/pull/1656

### CODE REVIEW

File: lib/plugins/aws/deploy/compile/events/apiGateway/tests/methods.js [hunk 1]
```diff
@@ -162,21 +162,25 @@ describe('#compileMethods()', () => {
...
-      awsCompileApigEvents.serverless.service.service}-${
-      awsCompileApigEvents.options.stage}-first/invocations`;
+    const lambdaUriObject = {
+      'Fn::Join': ['', [
+        'arn:aws:apigateway:',
+        { Ref: 'AWS::Region' },
+        ':lambda:path/2015-03-31/functions/',
+        { 'Fn::GetAtt': ['first', 'Arn'] },
+        '/invocations'  // eslint-disable-line comma-dangle
+      ]]                // eslint-disable-line comma-dangle
```
Comment (svdgraaf) [inline lib/plugins/aws/deploy/compile/events/apiGateway/tests/methods.js]: @pmuens I couldn't get eslint to parse this correctly, I'm not sure how to fix this correctly without disabling the eslint parser for these lines. Eslint wants me to add comma's here, but when I add them (which don't make sense), it wants me to delete them ;) Perhaps the notation is wrong?

Comment (pmuens) [inline lib/plugins/aws/deploy/compile/events/apiGateway/tests/methods.js]: You could try to write it in string representation (like we've done here: https://github.com/serverless/serverless/blob/v1.0/lib/plugins/aws/deploy/compile/events/schedule/index.js#L59-L71) and then parse it (https://github.com/serverless/serverless/blob/v1.0/lib/plugins/aws/deploy/compile/events/schedule/index.js#L86).

This way you could get rid of the linting errors without disabling the linter for those specific lines.

**Comment by pmuens** [comment]
Thank you for your contribution! Really looking forward to get this into place!

**Comment by svdgraaf** [comment]
@pmuens got it, fixed the json notation, now no more ugly disabling :) just let me know if I need to do a rebase whenever. Looking forward to the merge :)

**Comment by pmuens** [comment]
@svdgraaf awesome! Thank you! I'll let you know if this branch needs a rebase (otherwise we'll merge it directly).

**Comment by pmuens** [comment]
@svdgraaf We've just merged #1681 (global switch from `.yaml` to `.yml`).

Could you rebase your branch?

Thanks in advance! You rock!

**Comment by svdgraaf** [comment]
@pmuens done!

**Comment by eahefnawy** [comment]
Merging! Again, thanks a lot @svdgraaf 😊

**Pull request merged**